src/js/modules/core/services/local-settings/local-settings.service.js

```diff
@@ -10,20 +10,74 @@
     function LocalSettingsService($log, CONFIG, helperService, localStorageService) {
         var self = this;
 
+        var localSettingsDataSchema = {
+            isPhoneVerified: false,
+            phoneNumber: null,
+            phoneCountry: null,
+            phoneHash: null,
+
+
+            isContactsSynchronized: false,
+            // TODO Update pinOnOpen
+            /* PIN lock */
+            isPinOnOpen: false,            // ask for pin on each wallet open // TODO Change to true as default
+            pinFailureCount: 0,         // counter of pin input failures
+            pinLastFailure: null        // last pin input failure
+        };
+
         self._$log = $log;
         self._CONFIG = CONFIG;
         self._helperService = helperService;
 
+        // We only load from local storage once, after that this is set to TRUE
+        self._isReady = false;
+
         // Local settings
-        self._localSettingsStorage = localStorageService.init("localSettings", {
-            isSetupComplete: false,
-            isPhoneVerified: false,
-            isContactsSynchronized: false,
-            // TODO Use it here
-            isPinOnOpen: false
+        self._localSettingsStorage = localStorageService.init("localSettings", localSettingsDataSchema);
+
+        // Data object
+        self._localSettingsData = angular.extend({}, localSettingsDataSchema);
+
+        // Read only settings data object
+        // the object would be shared
+        self._readonlyDoc = {
+            readonly: true
+        };
+
+        angular.forEach(self._localSettingsData, function(value, key) {
```

Comment:
maybe move this read only doc stuff into the `localStorageService` ?

Reply:
Add to a ticket 

https://trello.com/c/nnQTOSsa/8-review-back-button-functionality